docs/use_cases/embeddings_on_browser.md

```diff
@@ -0,0 +1,628 @@
+5. **`modelComputing`** and **`setModelComputing`**:
+    - **`modelComputing`** is a state variable that tracks whether the model is currently computing embeddings.
+    - It is initially set to **`false`**.
+    - **`setModelComputing`** is a function to update the **`modelComputing`** state variable.
+6. **`showSimilarityMatrix`** and **`setShowSimilarityMatrix`**:
+    - **`showSimilarityMatrix`** is a state variable that controls the visibility of a similarity matrix in the user interface.
+    - It is initially set to **`false`**.
+    - **`setShowSimilarityMatrix`** is a function to update the **`showSimilarityMatrix`** state variable.
+7. **`embeddingModel`** and **`setEmbeddingModel`**:
+    - **`embeddingModel`** is a state variable that stores a TensorFlow.js tensor. It is used to represent the embeddings of sentences.
+    - It is initialized with a simple tensor with values **`[[1, 2], [3, 4]]`**.
+    - **`setEmbeddingModel`** is a function to update the **`embeddingModel`** state variable.
+8. **`canvasSize`** and **`setCanvasSize`**:
+    - **`canvasSize`** is a state variable representing the size of a canvas used to display the similarity matrix.
+    - It is initially set to **`0`**.
+    - **`setCanvasSize`** is a function to update the **`canvasSize`** state variable.
+9. **`similarityMatrix`** and **`setSimilarityMatrix`**:
+    - **`similarityMatrix`** is a state variable that stores the calculated similarity matrix for sentence embeddings.
+    - It is initially set to **`null`**.
+    - **`setSimilarityMatrix`** is a function to update the **`similarityMatrix`** state variable.
+
+```tsx
+// State variables to manage user input, loading state, and results
+  const [sentences, setSentences] = useState<string>('');
+  const [sentencesList, setSentencesList] = useState<string[]>([]);
+  const [embeddings, setEmbeddings] = useState<string>('');
+  const [modelLoading, setModelLoading] = useState(false);
+  const [modelComputing, setModelComputing] = useState(false);
+  const [showSimilarityMatrix, setShowSimilarityMatrix] = useState(false);
+  const [embeddingModel, setEmbeddingModel] = useState<tf.Tensor2D>(tf.tensor2d([[1, 2], [3, 4]]));
+  const [canvasSize, setCanvasSize] = useState(0);
+  const [similarityMatrix, setSimilarityMatrix] = useState<number[][] | null>(null);
+```
+
+## Function to toggle the display of the similarity matrix
+
+The **`handleSimilarityMatrix`** function toggles the display of a similarity matrix in the user interface by changing the **`showSimilarityMatrix`** state variable. If the matrix was previously shown, it hides it by setting it to **`null`**. If it wasn't shown, it calculates the matrix and sets it to be displayed in the user interface. This function is typically called when a user clicks a button or performs an action to show or hide the similarity matrix.
+
+1. **`const handleSimilarityMatrix = () => { ... }`**:
```

Comment:
same here, move comments to the code snippet (and only chose 3-4 key comments)

Reply:
happens in other places, I think the feedback is clear and I don't have to copy it everywhere